File: iteration2/CSE3063F21P1_GRP16/src/main/java/Models/E3HourCollusion.java

package Models;

import java.util.ArrayList;

public class E3HourCollusion  implements CourseEvents {
	
	
private ArrayList<Student> students;


public ArrayList<Student> getStudents() {
	return students;
}

public  E3HourCollusion() {
	students=new ArrayList<>();
	
}

@Override
public String toString() {

	String s="Total Number of Student who could't take the course becuase Hour Collusion: "+students.size()+"\nList Of Students:\n";
	
	for(int i=0;i<students.size();i++) {
		s+=students.get(i).getFirstName()+" "+students.get(i).getLastName()+"\n";
		
		
	}
	return s;
}

}

File: iteration3/CSE3063F21P1_GRP16/models/E3HourCollusion.py

from typing import List

from CourseEvents import CourseEvents

from Student import Student


class E3HourCollusion:
    def __init__(self):
        self.__students = []

    def __str__(self):
        s = "    Total Number of Student who couldn't take the course because of Hour Collusion: {}\n".format(len(self.__students))
        if len(self.__students) > 0:
            s += "     -List Of students:\n"
        else:
            s += "\n"

        for i in range(len(self.__students)):
            s += "      *{} {}\n".format(self.__students[i].getFirstName(), self.__students[i].getLastName())

        return s